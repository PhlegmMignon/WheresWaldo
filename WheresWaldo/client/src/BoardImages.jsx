import convention from "./assets/nMpQXwq.jpg";
import convention_kon from "./assets/kon.webp";
import convention_luffy from "./assets/luffy.png";
import convention_zim from "./assets/Zim.yelling.svg";

import stage2 from "./assets/stage3.png";
import stage2_fish from "./assets/fish.png";
import stage2_cannon from "./assets/cannon.png";
import stage2_egg from "./assets/egg.png";

export default function BoardImages() {
  const getImages = () => {
    return [
      {
        name: "convention",
        src: convention,
        id: 0,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
      },
      {
        name: "stage 2",
        src: stage2,
        id: 1,
        characters: [
          { name: "fish", src: stage2_fish, id: 0 },
          { name: "cannon", src: stage2_cannon, id: 1 },
          { name: "egg", src: stage2_egg, id: 2 },
        ],
      },
    ];
  };

  return { getImages };
}
