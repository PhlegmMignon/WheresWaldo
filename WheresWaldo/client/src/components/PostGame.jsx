import { useEffect, useState } from "react";
import Leaderboard from "./Leaderboard.jsx";
import ScorePrompt from "./ScorePrompt.jsx";

export default function PostGame({
  time,
  gameImage,
  setGameState,
  scores,
  getScores,
}) {
  const [isHighscore, setIsHighscore] = useState(false);

  useEffect(() => {
    detectHighScore(time, scores);
  }, []);

  function detectHighScore(time, scores) {
    let position;
    let seconds = time.minutes * 60 + time.seconds;

    if (scores.length < 5) {
      position = scores.length;
    } else {
      for (let i = 0; i < scores.length; i++) {
        if (seconds < scores[i].score) {
          position = i;
          break;
        }
      }
    }
    if (position != undefined) {
      setIsHighscore(true);
    }
  }

  return (
    <div id="imageContainer">
      <img id="image" src={gameImage.src} alt="picture of convention" />
      {isHighscore ? (
        <ScorePrompt
          setIsHighscore={setIsHighscore}
          time={time}
          gameImage={gameImage}
          getScores={getScores}
        />
      ) : (
        <Leaderboard scores={scores} setGameState={setGameState} />
      )}
    </div>
  );
}

//Questions:
//What would be a better way to handle detectHighscore and getScores?
