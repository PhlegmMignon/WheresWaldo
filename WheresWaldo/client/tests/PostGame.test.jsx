import { render } from "@testing-library/react";
import PostGame from "../src/components/PostGame.jsx";
import BoardImages from "../src/BoardImages.jsx";

describe("PostGame unit tests", () => {
  it("Renders postgame when highscore is false", async () => {
    let gameImage = BoardImages().getImages();

    const { container } = render(
      <PostGame
        isHighscore={false}
        gameImage={gameImage[0].src}
        scores={[
          { name: "asd", score: 5, _id: 0 },
          { name: "aaa", score: 5, _id: 1 },
          { name: "sss", score: 5, _id: 2 },
          { name: "ddd", score: 5, _id: 3 },
          { name: "zzz", score: 5, _id: 4 },
        ]}
        time={{ minutes: 2, seconds: 4 }}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("Renders postgame when highscore is true", async () => {
    let gameImage = BoardImages().getImages();

    const { container } = render(
      <PostGame
        isHighscore={false}
        gameImage={gameImage[0].src}
        scores={[{ name: "asd", score: 5, _id: 0 }]}
        time={{ minutes: 2, seconds: 4 }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
