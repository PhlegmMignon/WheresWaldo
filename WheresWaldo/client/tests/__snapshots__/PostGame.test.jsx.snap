// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostGame unit tests > Renders postgame when highscore is false 1`] = `
<div>
  <div
    id="imageContainer"
  >
    <img
      alt="picture of convention"
      id="image"
    />
    <div
      class="absolute inset-1/2 flex flex-col gap-3 h-72 w-60 rounded-xl p-6"
      id="leaderboard"
    >
      <div
        class="text-center text-lg "
      >
        Leaderboard
      </div>
      <div
        class="flex justify-between"
      >
        <p>
          Name
        </p>
         
        <p
          class="-ml-8"
        >
          Time
        </p>
      </div>
      <ol>
        <li
          class="flex justify-between"
        >
          <div>
            asd
          </div>
          <div>
            5
            secs
          </div>
        </li>
        <li
          class="flex justify-between"
        >
          <div>
            aaa
          </div>
          <div>
            5
            secs
          </div>
        </li>
        <li
          class="flex justify-between"
        >
          <div>
            sss
          </div>
          <div>
            5
            secs
          </div>
        </li>
        <li
          class="flex justify-between"
        >
          <div>
            ddd
          </div>
          <div>
            5
            secs
          </div>
        </li>
        <li
          class="flex justify-between"
        >
          <div>
            zzz
          </div>
          <div>
            5
            secs
          </div>
        </li>
      </ol>
      <button
        class="text-lg"
      >
        New game
      </button>
    </div>
  </div>
</div>
`;

exports[`PostGame unit tests > Renders postgame when highscore is true 1`] = `
<div>
  <div
    id="imageContainer"
  >
    <img
      alt="picture of convention"
      id="image"
    />
    <div
      class="absolute inset-1/2 flex flex-col gap-8 p-4 h-56 w-56 rounded-lg"
      id="scorePrompt"
    >
      <div
        class="text-center text-lg"
      >
        A new highscore!
      </div>
      <form
        action="/scores"
        class="flex flex-col gap-2 "
        method="post"
      >
        <label
          for="name"
        >
          Name
        </label>
        <input
          class="w-full"
          name="name"
          required=""
          type="text"
        />
        <div
          class="flex gap-14 mt-2.5 ml-7"
        >
          <button
            type="submit"
          >
            Submit
          </button>
          <button>
            Skip
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
